zebra-consensus/src/checkpoint.rs

```diff
@@ -56,10 +56,23 @@ struct QueuedBlock {
 
 /// A list of unverified blocks at a particular height.
 ///
-/// Typically contains zero or one blocks, but might contain more if a peer
+/// Typically contains a single block, but might contain more if a peer
 /// has an old chain fork. (Or sends us a bad block.)
+///
+/// The CheckpointVerifier avoids creating zero-block lists.
 type QueuedBlockList = Vec<QueuedBlock>;
 
+/// The maximum number of queued blocks at any one height.
+///
+/// This value is a tradeoff between:
+/// - rejecting bad blocks: higher is more efficient, and
+/// - avoiding a memory DoS: lower means less memory usage.
+///
+/// If we use a checkpoint interval of 500 blocks, our approximate queued block
+/// memory usage is:
+/// `500 blocks * 2 MB max per block * 2 blocks per height = 2 GB`
+const MAX_QUEUED_BLOCKS_PER_HEIGHT: usize = 2;
```

Comment:
I'm not sure it makes sense for this parameter to be different from the fanout parameter that we use in the syncer.  When we perform fanout to try to extend chain tips, we admit the possibility of up to `F` different tips.  So if there should be a limit here, it seems like the limit should be `F`, since that's the number of chain tips we're potentially pursuing in parallel.

Reply:
I made the fanout and blocks per height equal in 65ee727c8c3f87dc2ef539279684a0abb2d56dc8.